File: Pertemuan_12/Aplikasi_API/App Peminjaman/Peminjaman.php
<?php
//Simpanlah dengan nama file : Peminjaman.php
require_once 'database.php';
class Peminjaman 
{
    private $db;
    private $table = 'peminjaman';
    public $kode_pinjam = "";
    public $tgl_pinjam = "";
    public $tgl_kembali = "";
    public $kode_buku = "";
    public $kode_anggota = "";
    public $kode_petugas = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode_pinjam(int $kode_pinjam)
    {
        $query = "SELECT * FROM $this->table WHERE kode_pinjam = $kode_pinjam";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_pinjam`,`tgl_pinjam`,`tgl_kembali`,`kode_buku`,`kode_anggota`,`kode_petugas`) VALUES ('$this->kode_pinjam','$this->tgl_pinjam','$this->tgl_kembali','$this->kode_buku','$this->kode_anggota','$this->kode_petugas')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_pinjam = '$this->kode_pinjam', tgl_pinjam = '$this->tgl_pinjam', tgl_kembali = '$this->tgl_kembali', kode_buku = '$this->kode_buku', kode_anggota = '$this->kode_anggota', kode_petugas = '$this->kode_petugas' 
        WHERE id_peminjaman = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode_pinjam($kode_pinjam): int
    {
        $query = "UPDATE $this->table SET kode_pinjam = '$this->kode_pinjam', tgl_pinjam = '$this->tgl_pinjam', tgl_kembali = '$this->tgl_kembali', kode_buku = '$this->kode_buku', kode_anggota = '$this->kode_anggota', kode_petugas = '$this->kode_petugas' 
        WHERE kode_pinjam = $kode_pinjam";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id_peminjaman = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode_pinjam($kode_pinjam): int
    {
        $query = "DELETE FROM $this->table WHERE kode_pinjam = $kode_pinjam";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Pertemuan_12/Aplikasi_Client_API/Peminjaman.py
import requests
import json
class Peminjaman:
    def __init__(self):
        self.__id=None
        self.__kode_pinjam = None
        self.__tgl_pinjam = None
        self.__tgl_kembali = None
        self.__kode_buku = None
        self.__kode_anggota = None
        self.__kode_petugas = None
        self.__url = "http://localhost/appperpustakaan/peminjaman_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode_pinjam(self):
        return self.__kode_pinjam
        
    @kode_pinjam.setter
    def kode_pinjam(self, value):
        self.__kode_pinjam = value
    @property
    def tgl_pinjam(self):
        return self.__tgl_pinjam
        
    @tgl_pinjam.setter
    def tgl_pinjam(self, value):
        self.__tgl_pinjam = value
    @property
    def tgl_kembali(self):
        return self.__tgl_kembali
        
    @tgl_kembali.setter
    def tgl_kembali(self, value):
        self.__tgl_kembali = value
    @property
    def kode_buku(self):
        return self.__kode_buku
        
    @kode_buku.setter
    def kode_buku(self, value):
        self.__kode_buku = value
    @property
    def kode_anggota(self):
        return self.__kode_anggota
        
    @kode_anggota.setter
    def kode_anggota(self, value):
        self.__kode_anggota = value
    @property
    def kode_petugas(self):
        return self.__kode_petugas
        
    @kode_petugas.setter
    def kode_petugas(self, value):
        self.__kode_petugas = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode_pinjam(self, kode_pinjam):
        url = self.__url+"?kode_pinjam="+kode_pinjam
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_peminjaman']
            self.__kode_pinjam = item['kode_pinjam']
            self.__tgl_pinjam = item['tgl_pinjam']
            self.__tgl_kembali = item['tgl_kembali']
            self.__kode_buku = item['kode_buku']
            self.__kode_anggota = item['kode_anggota']
            self.__kode_petugas = item['kode_petugas']
        return data
    def simpan(self):
        payload = {
            "kode_pinjam":self.__kode_pinjam,
            "tgl_pinjam":self.__tgl_pinjam,
            "tgl_kembali":self.__tgl_kembali,
            "kode_buku":self.__kode_buku,
            "kode_anggota":self.__kode_anggota,
            "kode_petugas":self.__kode_petugas
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode_pinjam(self, kode_pinjam):
        url = self.__url+"?kode_pinjam="+kode_pinjam
        payload = {
            "kode_pinjam":self.__kode_pinjam,
            "tgl_pinjam":self.__tgl_pinjam,
            "tgl_kembali":self.__tgl_kembali,
            "kode_buku":self.__kode_buku,
            "kode_anggota":self.__kode_anggota,
            "kode_petugas":self.__kode_petugas
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode_pinjam(self,kode_pinjam):
        url = self.__url+"?kode_pinjam="+kode_pinjam
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text